arkane/encorr/reference.py

```diff
@@ -111,6 +112,12 @@ def __init__(self, species=None, smiles=None, adjacency_list=None, inchi=None, r
         if symmetry_number:
             self.symmetry_number = symmetry_number
 
+        # Reset to the provided identifiers to avoid translation errors
+        self.smiles = smiles or self.smiles
+        self.adjacency_list = adjacency_list or self.adjacency_list
+        self.inchi = inchi or self.inchi
+        self.inchi_key = inchi_key or self.inchi_key
```

Comment:
Assignment using `or` will assign the second value if the first one is falsy. Is that your desired behavior for all possible falsy values of the first variable in these assignments (e.g., if `smiles` is `''`)? If instead you wrote this as a shortcut for checking whether the first variable is `None` or not, then assignment using this `or` construct is discouraged, as comparisons against `None` should always be done using `is`. For example, `self.smiles = self.smiles if smiles is None else smiles`.

Alternatively, since `self.smiles` is already assigned prior to this, you can just do
```python
if smiles is not None:
    self.smiles = smiles
```
Potentially saves one assignment but adds an extra line. (Last alternative: since the expression is short, PEP 8 might allow `if smiles is not None: self.smiles = smiles` on a single line.)

Reply:
For documentation purposes, we tried the approach of `if smiles is not None: self.smiles = smiles`. However, we ran into the following issue during testing:

![image](https://user-images.githubusercontent.com/25854331/82843382-fdecb980-9eaa-11ea-90c4-d567ba0bab75.png)

We think that somewhere `''` is getting set as a SMILES string, which is being treated like a bool by RDKit

We decided to go back to `self.smiles = smiles or self.smiles`, because all of these identifiers should only be strings, and `''` is not a valid identifier whether it be a SMILES string, adjlist, or InChI. Therefore, we shouldn't run into issues where a valid identifier gets evaluated as False.

At some point we should track down where the smiles string default is being set as `''`